mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/search/definition/impl/PojoConstructorProjectionDefinition.java

```diff
@@ -79,6 +79,11 @@ public CompositeProjectionValueStep<?, T> apply(SearchProjectionFactory<?, ?> pr
 			return initialStep.from( innerProjections ).asArray( valueCreateHandle );
 		}
 		catch (SearchException e) {
+			if ( e.getCause() instanceof SearchException ) {
```

Comment:
It's a good idea to reduce verbosity.

However, this implementation is a bit dodgy, as it's theoretically possible for an exception where the context was not already added to have a cause that is a `SearchException`.

Maybe you should just use a dedicated subclass of `SearchException` when you add context, and test for that? A bit like we did in `org.hibernate.search.mapper.pojo.processing.impl.PojoIndexingProcessorPropertyNode#process`?